### PR TITLE
implement the replaceCharForBrowserConsistency function

### PR DESCRIPTION
**Pull request by neraliu**
this function replaces some known chars that will affect the parsing of the html5 page. for example, some version of IE will strip out the null character leading to vulnerability

example:
`<\x00script>alert(0);</script>`

the idea of this function is to replace some known chars to achieve browser implementation consistency

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: dist/secure-handlebars.js [hunk 1]
```diff
@@ -8848,7 +8879,7 @@ ContextParserHandlebars.prototype.analyzeContext = function(input) {
     var ast = this.buildAst(input, 0, []);
     var stateObj = this.getInternalState();
     var r = this.analyzeAst(ast, stateObj);
-    this._printChar(r.output);
+    this._printCharEnable? process.stdout.write(r.output) : '';
```
Comment (adon-at-work) [inline dist/secure-handlebars.js]: this._printCharEnable && process.stdout.write(r.output);

**Comment by yahoocla** [comment]
CLA is valid!

File: src/context-parser-handlebars.js [hunk 2]
```diff
@@ -241,7 +272,7 @@ ContextParserHandlebars.prototype.analyzeContext = function(input) {
     var ast = this.buildAst(input, 0, []);
     var stateObj = this.getInternalState();
     var r = this.analyzeAst(ast, stateObj);
-    this._printChar(r.output);
+    this._printCharEnable? process.stdout.write(r.output) : '';
```
Comment (adon-at-work) [inline src/context-parser-handlebars.js]: this._printCharEnable && process.stdout.write(r.output);

File: src/context-parser-handlebars.js [hunk 3]
```diff
@@ -249,12 +280,11 @@ ContextParserHandlebars.prototype.analyzeContext = function(input) {
...
-// @function module:ContextParserHandlebars._printChar
-ContextParserHandlebars.prototype._printChar = function(ch) {
-    if (this._printCharEnable) {
-        process.stdout.write(ch);
-    }
-    this._buffer.push(ch);
+// @function module:ContextParserHandlebars._saveToBuffer
+ContextParserHandlebars.prototype._saveToBuffer = function(str) {
+    /* it will not affect the state change */
+    str.length === 1? str = this.replaceCharForBrowserConsistency(str, -1) : '';
```
Comment (adon-at-work) [inline src/context-parser-handlebars.js]: str.length === 1 && (str = this.replaceCharForBrowserConsistency(str, -1));

File: src/context-parser-handlebars.js [hunk 4]
```diff
@@ -442,18 +472,18 @@ ContextParserHandlebars.prototype._consumeExpression = function(input, i, type,
         switch (type) {
             case handlebarsUtils.COMMENT_EXPRESSION_LONG_FORM:
                 if (input[j] === '-' && j+3<len && input[j+1] === '-' && input[j+2] === '}' && input[j+3] === '}') {
-                    saveToBuffer === true ? this._printChar('--}}') : obj.str += '--}}';
+                    saveToBuffer === true ? this._saveToBuffer('--}}') : obj.str += '--}}';
```
Comment (adon-at-work) [inline src/context-parser-handlebars.js]: can you batch replace all `saveToBuffer === true ?` with `saveToBuffer ?`

**Comment by neraliu** [comment]
the change is here as i just rename my repo to new repo name secure-handlebars.
https://github.com/neraliu/secure-handlebars/commit/3fc9ed43c26345a8a844cbb604f3a9475a535a23

**Pull request closed without merging**